cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/passes/SymbolResolver.kt

```diff
@@ -335,13 +342,8 @@ open class SymbolResolver(ctx: TranslationContext) : ComponentPass(ctx) {
         var type: Type = containingClass
 
         // Check for a possible overloaded operator->
-        if (
-            reference.language is HasOperatorOverloading &&
-                reference is MemberExpression &&
-                reference.operatorCode == "->" &&
-                reference.base.type !is PointerType
-        ) {
-            val result = resolveOperator(reference)
+        if (isOverloadedOperator(reference, "->")) {
+            val result = resolveOperator(reference as MemberExpression)
```

Comment:
Is it possible to get rid of the `as MemberExpression` which is ensured by the `isOverloadedOperator` function? The linter seems not to get this.

Reply:
Yes with a fancy Kotlin magic called contracts :)
```
contract { returns() implies (ex is MemberExpression) }
```